R/recipe_run.R

```diff
@@ -237,6 +238,17 @@ recipe_run <- function(info, parameters, envir, config, echo = TRUE) {
   hash_data_rds <- con_rds$mset(prep$data)
   stopifnot(identical(hash_data_csv, hash_data_rds))
 
+  ## make sure the resources have the same hashes (and size?)
+  post_run_resources <- get_resource_info(info)
+  stopifnot(identical(resource_info$hash_resources,
```

Comment:
These are not going to be clear enough - `stopifnot` should only be used for internal assertions and not for user facing errors.  Replace with an if block here, and explicitly test it (see test file for what needs doing there).  We'll back out the size changes for now, but in general if the size has changed the hash will change so that is sufficient.